.github/workflows/c-arduino-tests.yml

```diff
@@ -16,12 +16,16 @@ on:
       scheduler:
         required: false
         type: string
+      all-platforms:
+        required: false
+        default: true
+        type: boolean
 
 jobs:
   run:
     strategy:
       matrix:
-        platform: [ubuntu-latest, macos-latest]
+        platform: ${{ (inputs.all-platforms && fromJSON('["ubuntu-latest", "macos-latest"]')) || 'ubuntu-latest' }}
```

Comment:
I see you're using this pattern in most of the target-specific tests. What does it do, exactly?

Reply:
I copied it from here: https://github.com/actions/runner/issues/409